core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +109,38 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @param scm        the valid scm link
+     * @return folderPath if it valid or return the scm itself
+     */
+    private String searchPomFiles(Path directory, String pluginName, String scm) {
+        try (Stream<Path> paths = Files.find(directory, 2, (path, $) ->
+            path.getFileName().toString().equals("pom.xml")
+        )) {
+            return paths.filter(pom -> pomFileMatchesPlugin(pom, pluginName)).toString();
+        } catch (IOException e) {
+            LOGGER.error("Could not browse the folder during probe {}", pluginName, e);
+        }
+        return scm;
+    }
+
+    private boolean pomFileMatchesPlugin(Path pomFile, String pluginName) {
+        MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+        try (Reader reader = new InputStreamReader(new FileInputStream(pomFile.toFile()), StandardCharsets.UTF_8)) {
+            Model model = mavenReader.read(reader);
+            if (model.getPackaging().equals("hpi") && model.getArtifactId().equals(pluginName)) {
```

Comment:
when you do, `object.method()` you can get a `NullPointerException`, even more when you don't control de `object`. Even if in the current case it should be fine, you can protect yourself from this situation by writting 

```suggestion
            if ("hpi".equals(model.getPackaging()) && pluginName.equals(model.getArtifactId()) {
```